libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -132,13 +135,40 @@ const AP_Param::GroupInfo AP_Baro::var_info[] = {
     // Slot 12 used to be TEMP3
 #endif
 
+    // @Param: ADJ_RATE
+    // @DisplayName: baro adjustment rate
+    // @Description: Rate of barometer adjustment using GPS. If enabled (>=0.1), the barometer ground pressure is slowly adjusted with GPS altitude.  Recommended value: 1cm/sec
+    // @Units: cm/second
+    // @Range: 0 10
+    // @Increment: 0.1
+    AP_GROUPINFO("ADJ_RATE", 13, AP_Baro, _gps_adj_step, ADJ_RATE_DEFAULT),
+
+    // @Param: ADJ_TC
+    // @DisplayName: GPS filter time constant
+    // @Description: Time constant for the low pass filter applied to the GPS signal. As we only want to compensate for long-term trends, the should be fairly large (30sec...several mintes)
+    // @Units: seconds
+    // @Range: 10 600
+    // @Increment: 1
+    AP_GROUPINFO("ADJ_TC", 14, AP_Baro, _gps_adj_timeconstant, ADJ_TC_DEFAULT),
+
     AP_GROUPEND
 };
 
 /*
   AP_Baro constructor
  */
-AP_Baro::AP_Baro()
+AP_Baro::AP_Baro(AP_GPS &gps) :
+        _gps(&gps),
+        _gps_available(true)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    for (uint8_t i = 0; i < BARO_MAX_INSTANCES; i++) {
+        _gps_alt_error[i].set_cutoff_frequency(BARO_LOOP_FREQ, 1.f/float(_gps_adj_timeconstant.get()));
```

Comment:
- When the user runs a "last minute calibration" the same method void AP_Baro::update_calibration() is called and we do calibration of GPS reference there as well.
- The change in ground pressure or altitude offset is depreciated if GPS is available and has required precision. If there is no GPS or precision is more than the limit, then the calibration will happen and it is not affected.
Being a plane user myself, I also used these corrections to make adjustments but this PR eliminates the need to do so. It does automatic correction for that.